moped-database/migrations/1704744986000_substantial_completion_date/up.sql

```diff
@@ -0,0 +1,449 @@
+     LEFT JOIN moped_entity mel ON mel.entity_id = mp.project_lead_id
+     LEFT JOIN moped_proj_partners mpp2 ON mp.project_id = mpp2.project_id AND mpp2.is_deleted = false
+     LEFT JOIN moped_entity me2 ON mpp2.entity_id = me2.entity_id
+     LEFT JOIN work_activities on work_activities.project_id = mp.project_id
+     LEFT JOIN moped_users added_by_user ON mp.added_by = added_by_user.user_id
+     LEFT JOIN current_phase_view current_phase on mp.project_id = current_phase.project_id
+     LEFT JOIN moped_public_process_statuses mpps ON mpps.id = mp.public_process_status_id
+     LEFT JOIN child_project_lookup cpl on cpl.parent_id = mp.project_id
+     LEFT JOIN moped_proj_components_subtypes mpcs on mpcs.project_id = mp.project_id
+     LEFT JOIN LATERAL
+      (
+        SELECT mpn.project_note, mpn.date_created
+        FROM moped_proj_notes mpn
+        WHERE mpn.project_id = mp.project_id AND mpn.project_note_type = 2 AND mpn.is_deleted = false
+        ORDER BY mpn.date_created DESC
+        LIMIT 1
+      ) as proj_notes on true
+  WHERE
+    mp.is_deleted = false
+  GROUP BY
+    mp.project_id, 
+    mp.project_name, 
+    mp.project_description, 
+    ppll.project_team_members, 
+    mp.ecapris_subproject_id, 
+    mp.date_added,
+    mp.is_deleted,
+    me.entity_name,
+    mel.entity_name,
+    mp.updated_at,
+    mp.interim_project_id,
+    mp.parent_project_id,
+    mp.knack_project_id,
+    current_phase.phase_name,
+    current_phase.phase_key,
+    current_phase.phase_name_simple,
+    ptl.type_name,
+    mpcs.components,
+    fsl.funding_source_name,
+    added_by_user.first_name,
+    added_by_user.last_name,
+    mpps.name,
+    cpl.children_project_ids,
+    proj_notes.project_note,
+    proj_notes.date_created,
+    work_activities.contractors,
+    work_activities.contract_numbers,
+    work_activities.task_order_names,
+    work_activities.task_order_names_short,
+    work_activities.task_orders;
+
+-- add substantial_completion_date to agol view
+CREATE OR REPLACE VIEW component_arcgis_online_view AS (
+    SELECT 
+        mpc.project_id, 
+        comp_geography.project_component_id, 
+        comp_geography.feature_ids, 
+        mpc.component_id, 
+        comp_geography.geometry, 
+        comp_geography.line_geometry, 
+        comp_geography.signal_ids, 
+        council_districts.council_districts, 
+        comp_geography.length_feet_total, 
+        mc.component_name, 
+        mc.component_subtype, 
+        mc.component_name_full, 
+        subcomponents.subcomponents, 
+        work_types.work_types, 
+        component_tags.component_tags, 
+        mpc.description AS component_description, 
+        mpc.is_deleted is_project_component_deleted, 
+        plv.is_deleted is_project_deleted, 
+        mpc.interim_project_component_id, 
+        mpc.completion_date,
+        COALESCE(mpc.completion_date, substantial_completion_date) as substantial_completion_date,
```

Comment:
at the component level, the component's completion date always trumps the completion date from phasing.